plugins/performance-lab/includes/admin/rest-api.php

```diff
@@ -0,0 +1,201 @@
+ */
+function perflab_handle_activate_plugin( WP_REST_Request $request ) {
+	require_once ABSPATH . 'wp-admin/includes/plugin.php';
+	require_once ABSPATH . 'wp-admin/includes/plugin-install.php';
+	require_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
+	require_once ABSPATH . 'wp-admin/includes/class-wp-ajax-upgrader-skin.php';
+
+	// Require to make helper functions available.
+	require_once PERFLAB_PLUGIN_DIR_PATH . 'includes/admin/load.php';
+	require_once PERFLAB_PLUGIN_DIR_PATH . 'includes/admin/plugins.php';
+
+	$params = $request->get_json_params();
+
+	// Ensure the 'slug' parameter is present.
+	if ( ! isset( $params['slug'] ) ) {
+		return new WP_Error(
+			'missing_parameter',
+			__( 'Missing required parameter "slug".', 'performance-lab' ),
+			array( 'status' => 400 )
+		);
+	}
+
+	$plugin_slug = perflab_sanitize_plugin_slug( wp_unslash( $params['slug'] ) );
+	if ( null === $plugin_slug ) {
+		return new WP_Error(
+			'invalid_plugin',
+			__( 'Invalid plugin slug provided.', 'performance-lab' ),
+			array( 'status' => 400 )
+		);
+	}
+
+	// Install and activate the plugin and its dependencies.
+	$result = perflab_install_and_activate_plugin( $plugin_slug );
+	if ( is_wp_error( $result ) ) {
+		return new WP_Error(
+			'plugin_activation_failed',
+			$result->get_error_message(),
+			array( 'status' => 500 )
+		);
+	}
+
+	return new WP_REST_Response(
+		array(
+			'success' => true,
+		)
+	);
+}
+
+/**
+ * Handles REST API request to get plugin settings URL.
+ *
+ * @since n.e.x.t
+ * @access private
+ *
+ * @phpstan-param WP_REST_Request<array<string, mixed>> $request
+ *
+ * @param WP_REST_Request $request Request.
+ * @return WP_REST_Response|WP_Error Response.
+ */
+function perflab_handle_get_plugin_settings_url( WP_REST_Request $request ) {
+	require_once ABSPATH . 'wp-admin/includes/plugin.php';
+
+	// Require to make helper functions available.
+	require_once PERFLAB_PLUGIN_DIR_PATH . 'includes/admin/load.php';
+	require_once PERFLAB_PLUGIN_DIR_PATH . 'includes/admin/plugins.php';
+
+	$params = $request->get_json_params();
+
+	// Ensure the 'slug' parameter is present.
+	if ( ! isset( $params['slug'] ) ) {
+		return new WP_Error(
+			'missing_parameter',
+			__( 'Missing required parameter "slug".', 'performance-lab' ),
+			array( 'status' => 400 )
+		);
+	}
+
+	$plugin_slug = perflab_sanitize_plugin_slug( wp_unslash( $params['slug'] ) );
+	if ( null === $plugin_slug ) {
+		return new WP_Error(
+			'invalid_plugin',
+			__( 'Invalid plugin slug provided.', 'performance-lab' ),
+			array( 'status' => 400 )
+		);
+	}
+
+	$plugin_settings_url = perflab_get_plugin_settings_url( $plugin_slug );
+	if ( null === $plugin_settings_url ) {
+		return new WP_REST_Response(
+			array(
+				'success'           => true,
+				'pluginSettingsURL' => false,
+			)
+		);
+	}
+
+	return new WP_REST_Response(
+		array(
+			'success'           => true,
+			'pluginSettingsURL' => esc_url_raw( $plugin_settings_url ),
```

Comment:
I don't think sanitization of the URL is really necessary since it's not user data.

```suggestion
			'pluginSettingsURL' => $plugin_settings_url,
```